inc/aes_packages/namespace.php

```diff
@@ -0,0 +1,318 @@
+<?php
+/**
+ * AES Package File Integration.
+ *
+ * @package altis/cloud
+ */
+
+namespace Altis\Cloud\AES_Packages;
+
+use Altis;
+use Aws\ElasticsearchService\ElasticsearchServiceClient;
+use Exception;
+
+/**
+ * Hook into search module package support.
+ *
+ * @return void
+ */
+function setup() {
+	// Check the current environment and bail if it isn't AWS.
+	if ( ! in_array( Altis\get_environment_type(), [ 'development', 'staging', 'production' ], true ) ) {
+		return;
+	}
+
+	add_filter( 'altis.search.packages_dir', __NAMESPACE__ . '\\packages_dir', 9 );
+	add_filter( 'altis.search.create_package_id', __NAMESPACE__ . '\\create_package_id', 10, 4 );
+	add_filter( 'altis.search.get_package_id', __NAMESPACE__ . '\\get_package_id', 10, 2 );
+	add_action( 'altis.search.check_package_status', __NAMESPACE__ . '\\on_check_package_status', 10, 3 );
+	add_action( 'altis.search.dissociate_package', __NAMESPACE__ . '\\dissociate_package' );
+	add_action( 'altis.search.delete_package', __NAMESPACE__ . '\\delete_package' );
+	add_action( 'altis.search.deleted_package', __NAMESPACE__ . '\\on_deleted_package', 10, 4 );
+}
+
+/**
+ * Set the package upload directory to an S3 directory.
+ *
+ * @param string $path The directory to store packages in.
+ * @return string
+ */
+function packages_dir( string $path ) : string {
+	if ( ! defined( 'S3_UPLOADS_BUCKET' ) || empty( S3_UPLOADS_BUCKET ) ) {
+		return $path;
+	}
+
+	return sprintf( 's3://%s/es-packages', S3_UPLOADS_BUCKET );
+}
+
+/**
+ * Get Elasticsearch Service Client.
+ *
+ * @return ElasticsearchServiceClient
+ */
+function get_aes_client() : ElasticsearchServiceClient {
+	return Altis\get_aws_sdk()->createElasticsearchService( [
+		'version' => '2015-01-01',
+		'region' => Altis\get_environment_region(),
+	] );
+}
+
+/**
+ * Return the domain ID portion of the Elasticsearch Service host name.
+ */
+function get_elasticsearch_domain() : string {
+	return preg_replace( '#^search-([a-z][a-z0-9\-]+)-[a-z0-9]+\..*$#', '$1', 'search-platform-test-36c2d870-ylbu5ldvhmnb2blhnfiqxabzdq.eu-west-1.es.amazonaws.com' ); // ELASTICSEARCH_HOST );
+}
+
+/**
+ * Create the package on AES and return the package ID string.
+ *
+ * @param string|null $package_id The package file path for ES mappings.
+ * @param string $slug The package slug.
+ * @param string $file The package file path.
+ * @param bool $for_network Whether this is a network level package or not.
+ * @return string|null
+ */
+function create_package_id( ?string $package_id, string $slug, string $file, bool $for_network = false ) : ?string {
```

Comment:
IMO it would be better to allow `WP_Error` as a return, so the errors can be surfaced up the stack?

Reply:
Yeah the error handling sucks right now, I need to do some work on that part of it to better surface statuses and things in the UI.